### DISCUSSION TITLE
Newer releases no longer show up with Gem::SpecFetcher

### DISCUSSION THREAD

**Discussion by will** [sparklemotion/nokogiri#2893] [General Q&A]
I didn't want to open this up as an issue, since it's probably something wrong with rubygems, but I'm not sure.

Recent releases of nokogiri no longer show up in `Gem::SpecFetcher` starting with `v1.14.4`

```ruby
def find(ver)
  dep = Gem::Dependency.new("nokogiri",ver)
  Gem::SpecFetcher.new(Gem::SourceList.from(["https://rubygems.org"])).spec_for_dependency(dep, false)
end

versions = %w[
1.14.0 1.14.1 1.14.2 1.14.3 1.14.4 1.14.5
1.15.0 1.15.1 1.15.2
]

versions.each do |v|
  print "#{v}: "
  print find(v).first.size.to_s.rjust(2) rescue nil
  puts
end
```

```
1.14.0: 11
1.14.1: 11
1.14.2: 11
1.14.3: 11
1.14.4:  0
1.14.5:  0
1.15.0:  0
1.15.1:  0
1.15.2:  0
```

This also happens with sorbet-static, so it might be something that ruby gems changed with platform dependent gems? It's hard to tell. Other gems like `pg` don't seem to have the same problem. I haven't really exhaustively tested everything yet though.

I ran into this problem because I use nix to package/ci/develop my app and bundix has to find all the gems to write out nix derivations for them.

**Comment by flavorjones** [comment]
FWIW we haven't changed anything about our release process that would explain this. I'd suggest reporting this upstream in https://github.com/rubygems/rubygems !

**Reply by will** [reply]
Good idea, I'll do that. I'll leave this open and report back when/if I get an answer from over there.

**Comment by flavorjones** [comment]
Also FWIW I can't reproduce this on my machine:

```
$ ~/foo.rb 
1.14.0: 11
1.14.1: 11
1.14.2: 11
1.14.3: 11
1.14.4: 11
1.14.5: 11
1.15.0: 11
1.15.1: 11
1.15.2: 11

$ gem -v
3.4.10

$ ruby -v
ruby 3.2.2 (2023-03-30 revision e51014f9c0) [x86_64-linux]
```

**Reply by will** [reply]
Very interesting. It was a college of mine who first noticed that our update script was failing and I tracked it down to this, so it's happening on at least two completely different computers, and it happens for me on my mac and a linux vm I run on on the mac:

### linux nixos
```
[will@nixos:~]$ nix-shell -p ruby

[nix-shell:~]$ ruby -v
ruby 3.1.4p223 (2023-03-30 revision 957bb7cb81) [aarch64-linux]

[nix-shell:~]$ gem -v
3.4.12

[nix-shell:~]$ ruby noko.rb
1.14.0: 11
1.14.1: 11
1.14.2: 11
1.14.3: 11
1.14.4:  0
1.14.5:  0
1.15.0:  0
1.15.1:  0
1.15.2:  0
``` 

### macos, custom ruby
```
❯ ruby -v
ruby 3.2.2 (2023-03-30 revision e51014f9c0) [arm64-darwin22]
❯ gem -v
3.4.10
❯ ruby noko.rb
1.14.0: 11
1.14.1: 11
1.14.2: 11
1.14.3: 11
1.14.4:  0
1.14.5:  0
1.15.0:  0
1.15.1:  0
1.15.2:  0
```

### macos system-shipped ruby 
```
❯ ruby -v
ruby 2.6.10p210 (2022-04-12 revision 67958) [universal.arm64e-darwin22]
❯ gem -v
3.0.3.1
❯ ruby code/owl/noko.rb
1.14.0: 11
1.14.1: 11
1.14.2: 11
1.14.3: 11
1.14.4:  0
1.14.5:  0
1.15.0:  0
1.15.1:  0
1.15.2:  0
```
